src/scss/components/_button.scss

```diff
@@ -1,229 +1,75 @@
-/**
- * TODO: properly refactor these components, the styles are bonkers...
- */
-@use 'sass:math';
 @use 'microscope-sass/lib/bem';
 
-@import '~microscope-sass/lib/grid';
-@import '~microscope-sass/lib/typography';
-
 @import '@utrecht/components/dist/button/css/index.css';
 @import '@utrecht/components/dist/button-link/css/index.css';
 
-@import '../mixins/prefix';
-
-$button-line-height: $typography-line-height-text-big * $typography-font-size-text-big;
-$button-line-height-px: math.div($button-line-height, ($button-line-height * 0 + 1)) * 16px;
-$button-padding-v: math.div(($grid-row-height - $button-line-height-px), 2) -
-  $typography-size-border;
-$button-padding-h: $grid-margin-2;
-
-@mixin button-theme(
-  $name,
-  $color,
-  $color-background,
-  $color-border: var(--of-color-bg),
-  $color-background-hover: null,
-  $color-border-hover: null,
-  $color-active: null,
-  $color-background-active: null,
-  $color-border-active: null,
-  $force-colors: false,
-  $color-border-focus: null
-) {
-  // build the selector either as just the parent or the parent with modifier,
-  // based on the theme name
-  $sel: #{&};
-
-  @if $name != '' {
-    $sel: '#{&}--#{$name}';
-  }
-
-  // now output the actual CSS rules
-  @at-root #{$sel} {
-    @if $force-colors {
-      color: $color !important;
-      background-color: $color-background !important;
-    } @else {
-      color: $color;
-      background-color: $color-background;
-    }
-
-    // TODO: can be removed once there are utrecht design tokens for the bottom/right
-    // border color
-    border-right-color: $color-border !important;
-    border-bottom-color: $color-border !important;
-
-    &:focus,
-    &:focus-visible {
-      // TODO: can be removed once there are utrecht design tokens for the bottom/right border color
-      // issue: https: //github.com/nl-design-system/utrecht/issues/1406
-      border-color: $color-border-focus !important;
-      border-right-color: $color-border-focus !important;
-      border-bottom-color: $color-border-focus !important;
-      // TODO: can be removed once there are utrecht design tokens for the border-width
-      border-width: 2px !important;
-    }
-
-    &:not([aria-disabled='true']):hover {
-      @if $color-background-hover {
-        background-color: $color-background-hover;
-      }
-
-      @if $color-border-hover {
-        // TODO: can be removed once there are utrecht design tokens for the bottom/right
-        // border color
-        border-right-color: $color-border-hover !important;
-        border-bottom-color: $color-border-hover !important;
+// TODO: can be removed once there are utrecht design tokens for the bottom/right border color
+// issue: https: //github.com/nl-design-system/utrecht/issues/1406
+.utrecht-button {
```

Comment:
these styles now affect all themes - I'd instead opt for a simpler approach where we configure the design tokens for full borders and then apply CSS overrides scoped to our own theme:

```scss
.openforms-theme {
  .utrecht-button {
    border-block-start-color: transparent;
    border-inline-start-color: transparent;
  }
}
```

etc.